helpers/setup-sa.sh

```diff
@@ -17,73 +17,138 @@
+    p )
+      p=$OPTARG
+      ;;
+    b )
+      b=$OPTARG
+      ;;
+    f )
+      f=$OPTARG
+      ;;
+    : )
+      echo
+      echo " Error: option -${OPTARG} requires an argument"
+      usage
+      ;;
+   \? )
+      echo
+      echo " Error: invalid option -${OPTARG}"
+      usage
+      ;;
+  esac
+done
+shift $((OPTIND -1))
+
+# Check for required input variables
+if [ -z "${o}" ] || [ -z "${p}" ]; then
+  echo
+  echo " Error: -o <organization id> and -p <project id> required."
+  usage
 fi
 
 # Organization ID
 echo "Verifying organization..."
-ORG_ID="$(gcloud organizations list --format="value(ID)" --filter="$1")"
+ORG_ID="$(gcloud organizations list --format="value(ID)" --filter="$o")"
 
-if [[ $ORG_ID == "" ]];
-then
+if [[ $ORG_ID == "" ]]; then
   echo "The organization id provided does not exist. Exiting."
   exit 1;
 fi
 
- # Seed Project
+# Seed Project
 echo "Verifying project..."
-SEED_PROJECT="$(gcloud projects list --format="value(projectId)" --filter="$2")"
+SEED_PROJECT="$(gcloud projects list --format="value(projectId)" --filter="$p")"
 
-if [[ $SEED_PROJECT == "" ]];
-then
-   echo "The Seed Project does not exist. Exiting."
+if [[ $SEED_PROJECT == "" ]]; then
+  echo "The Seed Project does not exist. Exiting."
   exit 1;
 fi
 
 # Billing account
-if [ $# -eq 3 ]; then
+if [ -z "${b}" ]; then
+  echo "Skipping billing account verification... (parameter not passed)"
+else
   echo "Verifying billing account..."
-  BILLING_ACCOUNT="$(gcloud beta billing accounts list --format="value(ACCOUNT_ID)" --filter="$3")"
+  BILLING_ACCOUNT="$(gcloud beta billing accounts list --format="value(ACCOUNT_ID)" --filter="$b")"
 
-  if [[ $BILLING_ACCOUNT == "" ]];
-  then
+  if [[ $BILLING_ACCOUNT == "" ]]; then
     echo "The billing account does not exist. Exiting."
     exit 1;
   fi
+fi
+
+# Project Folder
+if [ -z "${f}" ]; then
+  echo "Skipping project folder verification... (parameter not passed)"
 else
-  echo "Skipping billing account verification... (parameter not passed)"
+  echo "Verifying project folder..."
+  FOLDER_ID="$(gcloud resource-manager folders list --format="value(ID)" --organization="$o" --filter="$f")"
+
+  if [[ $FOLDER_ID == "" ]]; then
+    echo "The project folder does not exist. Exiting."
+    exit 1;
+  fi
 fi
 
- # Seed Service Account creation
+# Seed Service Account creation
 SA_NAME="project-factory-${RANDOM}"
 SA_ID="${SA_NAME}@${SEED_PROJECT}.iam.gserviceaccount.com"
 STAGING_DIR="${PWD}"
 KEY_FILE="${STAGING_DIR}/credentials.json"
 
 echo "Creating Seed Service Account..."
 gcloud iam service-accounts \
-    --project "${SEED_PROJECT}" create ${SA_NAME} \
-    --display-name ${SA_NAME}
+  --project "${SEED_PROJECT}" create ${SA_NAME} \
+  --display-name ${SA_NAME}
 
 echo "Downloading key to credentials.json..."
 gcloud iam service-accounts keys create "${KEY_FILE}" \
-    --iam-account "${SA_ID}" \
+  --iam-account "${SA_ID}" \
+  --user-output-enabled false
+
+if [[ $FOLDER_ID == "" ]]; then
```

Comment:
This condition will never trigger based on the earlier exit [here](https://github.com/terraform-google-modules/terraform-google-project-factory/pull/295/files#diff-f3fdb2558703bcb409cbf6fdc6d5b5ecR114-R117).

Reply:
The first check occurs only if the "folder" parameter is passed, and exit if the project folder does not exist on gcp.
If the "folder" parameter is not passed, the earlier check is always skipped, and this check is necessary in order not to apply the permissions related to the folder.